src/Preg.php

```diff
@@ -25,8 +25,21 @@
 final class Preg
 {
     /**
-     * @param null|string[]         $matches
-     * @param int-mask<0, 256, 512> $flags
+     * @template TFlags as int-mask<0, 256, 512>
+     *
+     * @param null|mixed[] $matches
+     * @param TFlags       $flags
```

Comment:
Do we actually need templates here? Templates are required for inheritance/composition when child type is narrowed, in this case most probably `@param int-mask<0, 256, 512> $flags` is enough (see [here](https://phpstan.org/r/8ffb40c4-4846-48b3-ba0e-5b35e0a6f3b1)). Of course `$flags is 256` and so on should be used in conditional `@param-out`.

PS. same in `matchAll()`.

Reply:
Do you mean like this: https://github.com/6b7562617765726c6f73/PHP-CS-Fixer/commit/639279deeb762d8ab8b36ffb87182d7a496e7e86 ? Because it is not working.

Reply:
can you repro in https://phpstan.org/try? do you mean `@param-out` does not support conditional type based on `@param $var`?

Reply:
It seems so: https://phpstan.org/r/b8290ad1-7799-45ac-bbaa-a9d521421315

The difference in PHPDoc is only the change as in [here](https://github.com/6b7562617765726c6f73/PHP-CS-Fixer/commit/639279deeb762d8ab8b36ffb87182d7a496e7e86) - template `TFlags` vs variable `$flags` in `@param-out`.

Reply:
use template - https://github.com/phpstan/phpstan/issues/10699 :)

Reply:
Yeah, my example used `@return`, not `@param-out`, because I didn't thought it may be handled differently 😅. If it works with a template, then let's use the template, but also let's add an issue reference and `todo` for cleaning this later.

Reply:
@kubawerlos this is fixed now

Reply:
https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/pull/7882